runtime/engine/src/test/java/io/aklivity/zilla/runtime/engine/test/internal/binding/TestBindingFactory.java

```diff
@@ -275,6 +275,18 @@ private void onInitialBegin(
                             {
                                 doInitialReset(traceId);
                             }
+                            if (assertion.resolve)
+                            {
+                                String schema = handler.resolve(id);
+                                if (assertion.schema == null && schema != null)
+                                {
+                                    doInitialReset(traceId);
+                                }
+                                else if (assertion.schema != null && !assertion.schema.equals(schema))
+                                {
+                                    doInitialReset(traceId);
+                                }
+                            }
```

Comment:
Do we need to add the explicit concept of `resolve` as a boolean, or is it sufficient to derive that intent from `assertion.schema != null` instead ?

This perhaps seems more intuitive since it would read as asserting the id and schema, agree?